### PR TITLE
Fix configurers logic

### LINKED ISSUE #170: Custom `MeterRegistryConfigurers` lead to ClassCastException
### Task List

- [x] Steps to reproduce provided
- [x] Stacktrace (if present) provided
- [x] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

1. Checkout https://github.com/22dasi1bwi/micrometer-confgurer
2. Run the tests
3. See that `MicrometerConfigurerTest` is failing because the application could not start (ClassCastException)

### Stacktrace
```13:35:47.038 [ForkJoinPool.commonPool-worker-27] ERROR i.m.context.DefaultBeanContext - Error processing bean method Definition: io.micronaut.management.health.monitor.HealthMonitorTask.void monitor() with processor (io.micronaut.scheduling.processor.ScheduledMethodProcessor@2fec1da5): Error instantiating bean of type [io.micrometer.core.instrument.composite.CompositeMeterRegistry]: class io.micrometer.core.instrument.composite.CompositeMeterRegistry cannot be cast to class io.micrometer.statsd.StatsdMeterRegistry (io.micrometer.core.instrument.composite.CompositeMeterRegistry and io.micrometer.statsd.StatsdMeterRegistry are in unnamed module of loader 'app')
io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type [io.micrometer.core.instrument.composite.CompositeMeterRegistry]: class io.micrometer.core.instrument.composite.CompositeMeterRegistry cannot be cast to class io.micrometer.statsd.StatsdMeterRegistry (io.micrometer.core.instrument.composite.CompositeMeterRegistry and io.micrometer.statsd.StatsdMeterRegistry are in unnamed module of loader 'app')
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1927)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2647)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2633)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2305)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1223)
	at io.micronaut.context.UnresolvedProvider.get(UnresolvedProvider.java:51)
	at io.micronaut.configuration.metrics.binder.executor.ExecutorServiceMetricsBinder.onCreated(ExecutorServiceMetricsBinder.java:77)
	at io.micronaut.configuration.metrics.binder.executor.ExecutorServiceMetricsBinder.onCreated(ExecutorServiceMetricsBinder.java:48)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1954)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2647)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2633)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2305)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2279)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1245)
	at io.micronaut.context.AbstractBeanDefinition.getBeanForConstructorArgument(AbstractBeanDefinition.java:1013)
	at io.micronaut.scheduling.$ScheduledExecutorTaskSchedulerDefinition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1898)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:2647)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:2633)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2305)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2279)
	at io.micronaut.context.DefaultBeanContext.findBean(DefaultBeanContext.java:1266)
	at io.micronaut.context.DefaultBeanContext.findBean(DefaultBeanContext.java:744)
	at io.micronaut.scheduling.processor.ScheduledMethodProcessor.process(ScheduledMethodProcessor.java:104)
	at io.micronaut.context.DefaultBeanContext.lambda$null$30(DefaultBeanContext.java:1609)
	at java.base/java.util.concurrent.ForkJoinTask$RunnableExecuteAction.exec(ForkJoinTask.java:1426)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1020)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1656)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1594)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:183)
Caused by: java.lang.ClassCastException: class io.micrometer.core.instrument.composite.CompositeMeterRegistry cannot be cast to class io.micrometer.statsd.StatsdMeterRegistry (io.micrometer.core.instrument.composite.CompositeMeterRegistry and io.micrometer.statsd.StatsdMeterRegistry are in unnamed module of loader 'app')
	at com.example.StatsdConfigurer.supports(StatsdConfigurer.kt:8)
	at io.micronaut.configuration.metrics.micrometer.MeterRegistryFactory.compositeMeterRegistry(MeterRegistryFactory.java:61)
	at io.micronaut.configuration.metrics.micrometer.$MeterRegistryFactory$CompositeMeterRegistry0Definition.build(Unknown Source)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:1898)
	... 30 common frames omitted
```

### Expected Behaviour

I'm able to pass a type-argument to the interface named `MeterRegistryConfigurer` and that does not prevent the application from starting up. Furthermore, I want to be able to apply `MeterRegistryConfigurer`s for just a certain type of registry. Also, I would like to see the documentation under (https://micronaut-projects.github.io/micronaut-micrometer/latest/guide/#metricsConcepts) being fixed.

### Actual Behaviour

The application is failing while starting up, because of a ClassCastException.

### Environment Information

- **Operating System**: Arch Linux
- **Micronaut Version:** 2.1.2 (also tested with 2.0.2)
- **JDK Version:** 11

### Example Application

- https://github.com/22dasi1bwi/micrometer-confgurer

### Full description
The first example configuration in the documentation of micronaut-micrometer (https://micronaut-projects.github.io/micronaut-micrometer/latest/guide/#metricsConcepts) shows that no type-argument for `MeterRegistryConfigurer` is needed. However the interface requires a type-argument of type `<T extends MeterRegistry>`. 

Besides that, even if a type-argument would have been provided (let's say `StatsdMeterRegistry`), from my point of view that code wouldnt work either, because the passed instance in both methods `configure(meterRegistry: MeterRegistry)` and `supports(meterRegistry: MeterRegistry)` are always a `CompositeMeterRegistry` (therefore the ClassCastException at runtime).

That being said, I observed that Micronaut is always creating a `CompositeMeterRegistry` when metrics are globally enabled which is configured by all `MeterRegistryConfigurer`s provided. 
When I try to provide my own implementation of `MeterRegistryConfigurer` with a type-argument (let's say `StatsdMeterRegistry`) the application is not even starting up, because of a ClassCastException.

Where is the advantage of having that type-argument when it is not allowed to be anything else than **MeterRegistry** ?

### PR DESCRIPTION
**Pull request by alvarosanchez**
Fixes #170

### CODE REVIEW

File: micrometer-core/src/main/resources/logback.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        <encoder>
+            <pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %msg%n</pattern>
+        </encoder>
+    </appender>
+
+    <root level="info">
+        <appender-ref ref="STDOUT" />
+    </root>
+
+    <logger name="io.micronaut.configuration.metrics" level="TRACE"/>
```
Comment (graemerocher) [inline micrometer-core/src/main/resources/logback.xml:15]: This file shouldn't be in `src/main/resources`

File: micrometer-core/src/test/groovy/io/micronaut/configuration/metrics/management/endpoint/MetricsEndpointSpec.groovy [hunk 2]
```diff
@@ -190,7 +190,7 @@ class MetricsEndpointSpec extends Specification {
 
 
     @Unroll
-    void "test metrics endpoint get jvm details #name success with tags"() {
+    void "test metrics endpoint get jvm details success with tags"() {
```
Comment (graemerocher) [inline micrometer-core/src/test/groovy/io/micronaut/configuration/metrics/management/endpoint/MetricsEndpointSpec.groovy:193]: Is this change necessary?

**Review by graemerocher — CHANGES_REQUESTED** [review]

File: micrometer-core/src/main/resources/logback.xml (hunk 1 above)
Comment (alvarosanchez) [reply micrometer-core/src/main/resources/logback.xml:15]: Oops, my mistake

File: micrometer-core/src/test/groovy/io/micronaut/configuration/metrics/management/endpoint/MetricsEndpointSpec.groovy (hunk 2 above)
Comment (alvarosanchez) [reply micrometer-core/src/test/groovy/io/micronaut/configuration/metrics/management/endpoint/MetricsEndpointSpec.groovy:193]: Well, there isn't a `name` unroll variable, and the report was displaying `test metrics endpoint get jvm details #name success with tags`. It is a leftover from a copy/paste from other test.

But strictly speaking, it is not necessary
